br/pkg/lightning/config/config.go

```diff
@@ -204,9 +295,49 @@ type Lightning struct {
 	CheckRequirements bool   `toml:"check-requirements" json:"check-requirements"`
 	MetaSchemaName    string `toml:"meta-schema-name" json:"meta-schema-name"`
 
-	MaxError           MaxError `toml:"max-error" json:"max-error"`
-	MaxErrorRecords    int64    `toml:"max-error-records" json:"max-error-records"`
-	TaskInfoSchemaName string   `toml:"task-info-schema-name" json:"task-info-schema-name"`
+	MaxError MaxError `toml:"max-error" json:"max-error"`
+	// deprecated, use Conflict.MaxRecordRows instead
+	MaxErrorRecords    int64  `toml:"max-error-records" json:"max-error-records"`
+	TaskInfoSchemaName string `toml:"task-info-schema-name" json:"task-info-schema-name"`
+}
+
+// adjust assigns default values and check illegal values. The input TikvImporter
+// must be adjusted before calling this function.
+func (l *Lightning) adjust(i *TikvImporter) {
```

Comment:
```suggestion
func (l *Lightning) adjust(backend string) {
```

Reply:
But `Lightning.adjust()` dose not rely on `*TikvImporter`. Should it follow the rule you say yet?

Reply:
I have understood. If backend is invalid, it should be report an error when `importer.adjust()`. So, the dependency exists between them.